### PR TITLE
Fix segfaulting tests

### PR DESCRIPTION
**Pull request by mjcarroll**
# 🦟 Bug fix

## Summary

This removes the deprecated scene3d tests for `garden`.  These have been a source of flakiness for quite some time in CI.

Additionally, this removes the `unload` calls for the rendering engines, which can cause issues.

## Checklist
- [ ] Signed all commits for DCO
- [ ] Added tests
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [ ] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: test/integration/scene3d_load.cc
Comment (ahcorde) [inline test/integration/scene3d_load.cc:2]:
```suggestion
 * Copyright (C) 2022 Open Source Robotics Foundation
```

File: src/CMakeLists.txt
Comment (ahcorde) [inline src/CMakeLists.txt:19]: This changes are related to this PR https://github.com/gazebosim/gz-gui/pull/428

File: test/integration/scene3d_config.cc
Comment (ahcorde) [inline test/integration/scene3d_config.cc:2]:
```suggestion
 * Copyright (C) 2022 Open Source Robotics Foundation
```

File: test/integration/scene3d_config.cc
Comment (ahcorde) [inline test/integration/scene3d_config.cc:51]: `joinPaths`

File: test/integration/scene3d_load.cc
Comment (ahcorde) [inline test/integration/scene3d_load.cc:51]: `joinPaths`

File: test/integration/transport_scene_manager.cc
Comment (ahcorde) [inline test/integration/transport_scene_manager.cc:264]: remove ?

File: test/integration/transport_scene_manager.cc
Comment (ahcorde) [inline test/integration/transport_scene_manager.cc:23]: is this required?

**Review by ahcorde — CHANGES_REQUESTED** [review]

File: test/integration/scene3d_config.cc
Comment (mjcarroll) [reply test/integration/scene3d_config.cc:2]: I'm going to keep the date the same, since it's just a piece of the former file.

File: test/integration/transport_scene_manager.cc
Comment (mjcarroll) [reply test/integration/transport_scene_manager.cc:23]: No, it's part of other work, will remove for now.

File: test/integration/transport_scene_manager.cc
Comment (ahcorde) [inline test/integration/transport_scene_manager.cc:263]: ?

File: test/integration/transport_scene_manager.cc
Comment (ahcorde) [reply test/integration/transport_scene_manager.cc:23]: it's still there

**Review by ahcorde — CHANGES_REQUESTED** [review]

File: test/integration/transport_scene_manager.cc
Comment (mjcarroll) [reply test/integration/transport_scene_manager.cc:263]: Unloading the scene is what causes the issues.

**Review by ahcorde — CHANGES_REQUESTED** [review]
conflicts

**Comment by iche033** [comment]
follow up PR in https://github.com/gazebosim/gz-gui/pull/535

File: src/plugins/minimal_scene/MinimalScene.hh
Comment (iche033) [inline src/plugins/minimal_scene/MinimalScene.hh:216]: maybe we should change the default in harmonic so this does not cause a behavior change?

Comment (azeey) [reply src/plugins/minimal_scene/MinimalScene.hh:216]: Agreed. gz-sim defaults to ogre2 through its default gui.config so no harm in waiting till Harmonic for this.

**Review by azeey — APPROVED** [review]
Just one minor issue. It'd be best if we merge #535 into this before merging this.

**Comment by azeey** [comment]
@iche033 is this good to go?

File: src/plugins/minimal_scene/MinimalScene.hh
Comment (mjcarroll) [reply src/plugins/minimal_scene/MinimalScene.hh]: This is now defaulting to "ogre" in this branch.

**Comment by mjcarroll** [comment]
I think this is good now, going to wait for CI to turn over, first.

**Comment by mjcarroll** [comment]
Windows is backed up, so I'm going without.

**Pull request merged**
